Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md

```diff
@@ -0,0 +1,137 @@
+---
+versionFrom: 7.0.0
+---
+
+# Adding a type to the provider model
+
+To add a new type, no matter if it's a workflow, field, data source, etc, there is a number of tasks to perform to connect to the Forms provider model. This chapter walks through each step and describes how each part works. This chapter will reference the creation of a workflow type. It is, however, the same process for all types.
+
+## Preparations
+
+Create a new ASP.NET or class project in Visual Studio add references to the Umbraco.Forms.Core.dll.
+
+## Adding the type to Forms
+
+The Forms API contains a collection of classes that the provider model automatically registers. So to add a new type to Forms you inherit from the right class. In the sample below we use the class for the workflow type.
+
+```csharp
+public class Class1 : Umbraco.Forms.Core.WorkflowType
```

Comment:
Can we have a better name please than `Class1` @doogstar

Reply:
Renamed as suggested, sorry! :)